llvm/lib/Analysis/DomConditionCache.cpp

```diff
@@ -34,23 +34,39 @@ static void findAffectedValues(Value *Cond,
     }
   };
 
-  ICmpInst::Predicate Pred;
-  Value *A;
-  if (match(Cond, m_ICmp(Pred, m_Value(A), m_Constant()))) {
-    AddAffected(A);
+  bool TopLevelIsAnd = match(Cond, m_LogicalAnd());
+  SmallVector<Value *, 8> Worklist;
+  SmallPtrSet<Value *, 8> Visited;
+  Worklist.push_back(Cond);
+  while (!Worklist.empty()) {
```

Comment:
What is the compile-time impact after this patch?

Reply:
http://llvm-compile-time-tracker.com/compare.php?from=38476b063f164995b85e47472e3c2e0a9c5f9075&to=bae764a404d1fed0a844dcbd74d0ffebd74735ff&stat=instructions:u